src/main/scala/temple/generate/database/DatabaseGenerator.scala

```diff
@@ -3,7 +3,8 @@ package temple.generate.database
 import temple.generate.database.ast.Statement
 
 /** DatabaseGenerator provides an interface for generating DB specific query languages from an AST */
-trait DatabaseGenerator[Context <: DatabaseContext] {
+trait DatabaseGenerator {
+  type Context <: DatabaseContext
```

Comment:
Why?

Reply:
@xsanda 

This comes from previously wanting to pass a `PostgresGenerator` around to be used inside the Go generation to generate Postgres. Decided against that approach, but Charlie thought this should still be like this.

Reply:
The correct context needs to be passed around for the correct generator, and having this as a parameter wasn’t as flexible as having it as a type argument (there was some weird contravariant type error)

Reply:
It means there is a type called `Context`, defined in the concrete implementations, which is a subtype of `DatabaseContext`.

Reply:
Seems familiar https://www.hackingwithswift.com/articles/74/understanding-protocol-associated-types-and-their-constraints :P